patches/minecraft/net/minecraftforge/fml/common/registry/GameData.java.patch

```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraftforge/fml/common/registry/GameData.java
++++ ../src-work/minecraft/net/minecraftforge/fml/common/registry/GameData.java
+@@ -837,6 +837,8 @@
+ 
+         for (IBlockState state : ((List<IBlockState>)block.func_176194_O().func_177619_a()))
+         {
++            if (state instanceof net.minecraftforge.common.property.IExtendedBlockState)
++                state = ((net.minecraftforge.common.property.IExtendedBlockState) state).getClean();
```

Comment:
In theory get listed states should ALWAYS return the clean states, but I guess it doesn't hurt to be safe.
But this needs to not be a patch it needs to be in the FML code base itself.